sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala

```diff
@@ -894,8 +894,8 @@ case class MapFromEntries(child: Expression)
 case class MapSort(base: Expression)
   extends UnaryExpression with NullIntolerant with QueryErrorsBase {
 
-  val keyType: DataType = base.dataType.asInstanceOf[MapType].keyType
-  val valueType: DataType = base.dataType.asInstanceOf[MapType].valueType
+  private lazy val keyType: DataType = base.dataType.asInstanceOf[MapType].keyType
+  private lazy val valueType: DataType = base.dataType.asInstanceOf[MapType].valueType
```

Comment:
why lazy val? it's a very cheap operation

Reply:
The stroy is that, it's a bit risk to call `dataType` when constructing expr since the child may be unresolved. I face that issue when adding test in catalyst. But now those tests are removed.. I just leave them lazy.